src/inmanta/moduletool.py

```diff
@@ -254,10 +248,10 @@ def diff(cls, *, low: Version, high: Version) -> Optional["ChangeType"]:
         if high.micro > low.micro:
             return cls.PATCH
         if len(high.base_version.split('.')) >= 4:
-            high_revision = high.base_version.split('.')[-1]
+            high_revision = high.base_version.split('.')[3]
             # We are switching from 3 digits to 4
             if len(low.base_version.split('.')) < 4 or \
-                (len(low.base_version.split('.')) >= 4 and high_revision > low.base_version.split('.')[-1]):
+                (len(low.base_version.split('.')) >= 4 and high_revision > low.base_version.split('.')[3]):
```

Comment:
This can still be simplified by dropping the redundant `len(low.base_version.split('.')) >= 4`.